src/Sulu/Bundle/AdminBundle/DependencyInjection/Compiler/ContentNavigationPass.php

```diff
@@ -10,34 +10,38 @@
 
 namespace Sulu\Bundle\AdminBundle\DependencyInjection\Compiler;
 
-use Sulu\Bundle\AdminBundle\Navigation\ContentNavigationInterface;
+use Sulu\Bundle\AdminBundle\Navigation\ContentNavigationProviderInterface;
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 
 /**
  * Add all services with given tag to the bundle content navigation
  */
-abstract class ContentNavigationPass implements CompilerPassInterface
+class ContentNavigationPass implements CompilerPassInterface
 {
-    protected $tag = null;
-    protected $serviceName = null;
-
     /**
      * {@inheritDoc}
      */
     public function process(ContainerBuilder $container)
     {
-        if (null !== $this->tag && null !== $this->serviceName && $container->hasDefinition($this->serviceName)) {
-            $contentNavigation = $container->getDefinition($this->serviceName);
-
-            $taggedServices = $container->findTaggedServiceIds($this->tag);
+        $contentNavigationCollector = $container->getDefinition('sulu_admin.content_navigation_collector');
 
-            foreach ($taggedServices as $id => $attributes) {
-                /** @var ContentNavigationInterface $navigation */
-                $navigation = $container->getDefinition($id);
+        $taggedServices = $container->findTaggedServiceIds('sulu.admin.content_navigation');
```

Comment:
I rename it, and sometimes later we should also change the `sulu.admin` tag to `sulu_admin.admin`?